hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/AbstractIdentifiableType.java

```diff
@@ -423,8 +424,10 @@ else if ( id != null ) {
 				final EmbeddableDomainType<?> compositeType = (EmbeddableDomainType<?>) type;
 				return new EmbeddedSqmPathSource<>(
 						EntityIdentifierMapping.ROLE_LOCAL_NAME,
+						(SqmPathSource) id,
 						compositeType,
-						Bindable.BindableType.SINGULAR_ATTRIBUTE
+						Bindable.BindableType.SINGULAR_ATTRIBUTE,
+						false
```

Comment:
Is this the `generic` flag? Is this correct?

Reply:
Nope, with `isGeneric()` on `SqmPathSource' (which is great IMO) I could pass `id.isGeneric()` instead of false which should be correct.